--- mock_vp.c
+++ mock_vp.c
@@ -1,6 +1,6 @@
+
 #include <stdio.h>
 #include <string.h>
-#include <unistd.h>
 
 char items[3][10];
 
@@ -11,31 +11,28 @@
         printf("input item:");
         buff = &items[i][0];
         i++;
-        fgets(buff, 40, stdin);
-        buff[strcspn(buff, "\n")] = 0;
-    }while(strlen(buff)!=0);
-    i--;
+        scanf("%s", buff);
+    } while (i < 3);
 }
 
 void func_b(){
     char *buff;
-    printf("done adding items\n");
+    printf("done adding items");
     int j;
     printf("display item #:");
     scanf("%d", &j);
-    buff = &items[j][0];
-    printf("item %d: %s\n", j, buff);
+    if (j >= 0 && j < 3) {
+        buff = &items[j][0];
+        printf("item %d: %s", j, buff);
+    } else {
+        printf("Invalid item number. Please enter a value between 0 and 2.");
+    }
 }
 
 #ifndef ___TEST___
-int main()
-{
-
+int main() {
     func_a();
-
     func_b();
-
-
     return 0;
 }
 #endif
